### DISCUSSION TITLE
Automatic baggage propagation

### DISCUSSION THREAD

**Discussion by stanosamek** [open-telemetry/opentelemetry-java-instrumentation#4518] [Q&A] [answered]
Hi,

We have the same problem as was discussed here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/4351?sort=old.  We would like to add baggage entries as span attributes to into every spans and we would like to use javaagents. 

Is it neccessary to build some extensions to provide baggage entries into spans as attributes? Shouldn't be in configuration some switch for turning this feature or not?

Thank you

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @stanosamek ,
We currently don't have any configuration switch that could do that; you have to build a custom extension for that.

**Reply by stanosamek** [reply]
Is it in roadmap to add this kind of feature?

**Reply by mateuszrzeszutek** [reply]
Hmm, we didn't really think of it; but it's not the first time this topic arises, I believe. Can you [create an enhancement issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/new/choose) and describe your problem there?
Thanks!

**Reply by stanosamek** [reply]
Ok, I'll create an enhancement issue.
